homeassistant/util/yaml/loader.py

```diff
@@ -275,6 +275,9 @@ def _load_secret_yaml(secret_path: str) -> JSON_TYPE:
 
 def secret_yaml(loader: SafeLineLoader, node: yaml.nodes.Node) -> JSON_TYPE:
     """Load secrets and embed it into the configuration YAML."""
+    if os.path.basename(loader.name) == SECRET_YAML:
+        _LOGGER.error("secrets.yaml: attempt to load secret from within secrets file")
+        return ""
```

Comment:
I think that we should raise, not silently fail.

Reply:
Done